### PR TITLE
feat: environment variables via the kernel arguments

### LINKED ISSUE #6984: HTTP proxy variables for kernel arguments
## Feature Request

It would be helpful to add extra kernel command-line arguments so that Talos Linux is able to use an http proxy at the boot process before obtaining the `talos.config` content and before the network is configured.

Potentially, the variables could be named `http_proxy` and `https_proxy`.

### Description

In most of our environments egress capabilities are restricted, particularly during machine provisioning, where the out-of-band network that is used to fetch the boot artifact and boot the machine, also prevents communication to the outside world. It's not possible to reach out to any external http services, whether on the internet or even outside of the server failure domain.

We currently have to store the talos config content on a static file server located in the same network where the machine is located. To use Omni, it means having a L4 proxy server located in that environment that will proxy the requests towards the Omni endpoint, which we have to deploy in each failure domain.

### PR DESCRIPTION
**Pull request by smira**
Unify getting environment variables, support passing environment variables via kernel args.

Fixes #6984
See #6999

For META this will be used to pass environment variables to the installer for ISO images (or PXE booting).

### CODE REVIEW

**Review by frezbo — APPROVED** [review]

File: internal/pkg/environment/environment.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+	for {
+		val := param.Get(idx)
+		if val == nil {
+			break
+		}
+
+		result = append(result, *val)
+		idx++
+	}
```
Comment (DmitriyMV) [inline internal/pkg/environment/environment.go:40]:
```suggestion
	for idx := 0; ; idx++ {
		val := param.Get(idx)
		if val == nil {
			break
		}
		result = append(result, *val)
	}
```

File: internal/pkg/environment/environment.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+	if cfg != nil {
+		for k, v := range cfg.Machine().Env() {
+			result = append(result, fmt.Sprintf("%s=%v", k, v))
```
Comment (DmitriyMV) [inline internal/pkg/environment/environment.go:44]:
```suggestion
			result = append(result, k+"="+v)
```

This should do the trick.

**Review by DmitriyMV — APPROVED** [review]
Few minor nits, but all good otherwise.

File: internal/pkg/environment/environment.go (hunk 1 above)
Comment (smira) [reply internal/pkg/environment/environment.go]: thanks, fixed!

File: internal/pkg/environment/environment.go (hunk 2 above)
Comment (smira) [reply internal/pkg/environment/environment.go]: thanks, fixed!

**Pull request merged**